<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>875</width>
    <height>615</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GlintMaskGenerator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QFrame" name="frame_13">
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="1">
        <widget class="QWidget" name="widget_5" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QRadioButton" name="img_type_rgb_radio">
            <property name="text">
             <string>RGB</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="img_type_cir_radio">
            <property name="text">
             <string>CIR</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="img_type_p4ms_radio">
            <property name="text">
             <string>P4MS</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="img_type_micasense_radio">
            <property name="text">
             <string>MicasenseRedEdge</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Imagery type</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="DirectoryPath" name="mask_dir_w" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="DirectoryPath" name="img_dir_w" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Output directory</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Image directory</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QFrame" name="frame_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_13">
          <item>
           <widget class="QRadioButton" name="method_threshold_radio">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Band threshold</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="method_ratio_radio">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Intensity ratio</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Mask method</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="parameters_stack">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="threshold_params_widget">
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Band thresholds (more masking &lt;---&gt; less masking)</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="7" column="2">
            <widget class="QWidget" name="widget_4" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="reset_thresholds_btn">
                <property name="text">
                 <string>Reset all</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>NIR</string>
             </property>
             <property name="buddy">
              <cstring>nir_thresh_w</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Blue</string>
             </property>
             <property name="buddy">
              <cstring>blue_thresh_w</cstring>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Green</string>
             </property>
             <property name="buddy">
              <cstring>green_thresh_w</cstring>
             </property>
            </widget>
           </item>
           <item row="6" column="2">
            <widget class="ThresholdCtrl" name="nir_thresh_w" native="true"/>
           </item>
           <item row="2" column="2">
            <widget class="ThresholdCtrl" name="red_thresh_w" native="true"/>
           </item>
           <item row="4" column="2">
            <widget class="ThresholdCtrl" name="rededge_thresh_w" native="true"/>
           </item>
           <item row="0" column="2">
            <widget class="ThresholdCtrl" name="blue_thresh_w" native="true"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Red Edge</string>
             </property>
             <property name="buddy">
              <cstring>rededge_thresh_w</cstring>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>Red</string>
             </property>
             <property name="buddy">
              <cstring>red_thresh_w</cstring>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="ThresholdCtrl" name="green_thresh_w" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_12">
          <property name="title">
           <string>Pixel buffer radius</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="BufferCtrl" name="pixel_buffer_w" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="ratio_params_widget">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>GroupBox</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>GroupBox</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>76</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QWidget" name="widget_3" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>805</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Max Workers</string>
         </property>
         <property name="buddy">
          <cstring>max_workers_spinbox</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="max_workers_spinbox">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget_14" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QProgressBar" name="progress_bar">
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="run_btn">
         <property name="text">
          <string>Run</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ThresholdCtrl</class>
   <extends>QWidget</extends>
   <header>widgets/threshold_ctrl.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DirectoryPath</class>
   <extends>QWidget</extends>
   <header>widgets/directory_path.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>BufferCtrl</class>
   <extends>QWidget</extends>
   <header>widgets/buffer_ctrl.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>img_type_rgb_radio</tabstop>
  <tabstop>img_type_p4ms_radio</tabstop>
  <tabstop>img_type_micasense_radio</tabstop>
  <tabstop>method_threshold_radio</tabstop>
  <tabstop>method_ratio_radio</tabstop>
  <tabstop>run_btn</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>run_btn</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>run_btn_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>984</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>954</x>
     <y>614</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>method_threshold_radio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>update_method_params()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>452</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>method_ratio_radio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>update_method_params()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>974</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>671</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset_thresholds_btn</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset_thresholds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>932</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>998</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>img_type_rgb_radio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>enable_available_thresholds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>281</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>img_type_cir_radio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>enable_available_thresholds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>370</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>img_type_p4ms_radio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>enable_available_thresholds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>579</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>589</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>img_type_micasense_radio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>enable_available_thresholds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>836</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>846</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>img_dir_btn_clicked()</slot>
  <slot>mask_dir_btn_clicked()</slot>
  <slot>run_btn_clicked()</slot>
  <slot>preview_btn_clicked()</slot>
  <slot>update_method_params()</slot>
  <slot>reset_thresholds()</slot>
  <slot>enable_available_thresholds()</slot>
 </slots>
</ui>
